diff --git a/plugin/megaraid_plugin/megaraid.py b/plugin/megaraid_plugin/megaraid.py
index 748bdba..0210235 100644
--- a/plugin/megaraid_plugin/megaraid.py
+++ b/plugin/megaraid_plugin/megaraid.py
@@ -791,7 +791,7 @@ def volume_raid_info(self, volume, flags=Client.FLAG_RSVD):
                 int(vd_prop_info['Span Depth']))
         elif raid_type == Volume.RAID_TYPE_RAID10:
             strip_count = (
-                int(vd_prop_info['Number of Drives Per Span']) / 2 *
+                int_div(int(vd_prop_info['Number of Drives Per Span']), 2) *
                 int(vd_prop_info['Span Depth']))
         else:
             # MegaRAID does not support 15 or 16 yet.
